tests/coreutils/date.rs

```diff
@@ -0,0 +1,712 @@
+use std::ffi::OsString;
+use uutils_args::{Arguments, Options, Value};
+
+// Note: "+%s"-style format options aren't covered here, but should be!
+
+// +%s
+//   -I[FMT], --iso-8601[=FMT]  output date/time in ISO 8601 format.
+//   -R, --rfc-email            output date and time in RFC 5322 format.
+//       --rfc-3339=FMT         output date/time in RFC 3339 format.
+// date, hours, minutes, seconds, ns
+// date, seconds, ns
+
+#[derive(Default, Debug, PartialEq, Eq, Value)]
+enum Iso8601Format {
+    #[default]
+    #[value("date")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("dat")]
+    #[value("da")]
+    #[value("d")]
+    Date,
+
+    #[value("hours")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("hour")]
+    #[value("hou")]
+    #[value("ho")]
+    #[value("h")]
+    Hours,
+
+    #[value("minutes")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("minute")]
+    #[value("minut")]
+    #[value("minu")]
+    #[value("min")]
+    #[value("mi")]
+    #[value("m")]
+    Minutes,
+
+    #[value("seconds")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("second")]
+    #[value("secon")]
+    #[value("seco")]
+    #[value("sec")]
+    #[value("se")]
+    #[value("s")]
+    Seconds,
+
+    #[value("ns")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("n")]
+    Ns,
+}
+
+#[derive(Debug, PartialEq, Eq, Value)]
+enum Rfc3339Format {
+    #[value("date")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("dat")]
+    #[value("da")]
+    #[value("d")]
+    Date,
+
+    #[value("seconds")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("second")]
+    #[value("secon")]
+    #[value("seco")]
+    #[value("sec")]
+    #[value("se")]
+    #[value("s")]
+    Seconds,
+
+    #[value("ns")]
+    // TODO: Express the concept "accepts prefixes" more nicely.
+    #[value("n")]
+    Ns,
+}
+
+#[derive(Arguments)]
+enum Arg {
+    #[arg("-I[FMT]")]
+    #[arg("--iso-8601[=FMT]")]
+    Iso(Iso8601Format),
+
+    #[arg("--rfc-3339=FMT")]
+    Rfc3339(Rfc3339Format),
+
+    #[arg("-R")]
+    #[arg("--rfc-email")]
+    RfcEmail,
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+enum Format {
+    #[default]
+    Unspecified,
+    Iso8601(Iso8601Format),
+    Rfc3339(Rfc3339Format),
+    RfcEmail,
+    // FromString(OsString),
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+struct Settings {
+    chosen_format: Format,
+}
+
+const MAGIC_MULTI_OUTPUT_ARG: &str = "! multiformat";
+
+impl Options<Arg> for Settings {
+    fn apply(&mut self, arg: Arg) -> Result<(), uutils_args::Error> {
+        if self.chosen_format != Format::Unspecified {
```

Comment:
So, I take a bit of issue with this example, because I don't like how `Unspecified` is a special thing that only exists for parsing. Essentially, it comes down to this: https://github.com/uutils/uutils-args/issues/11.

BUT! I don't care enough to block this. We can improve this later and we should move towards shipping this crate.